js/AssessmentSet.js

```diff
@@ -347,11 +349,11 @@ export default class AssessmentSet extends ScoringSet {
   }
 
   /**
-   * Returns whether the set contains any components configured to "soft" reset
+   * Returns whether all components are configured to "soft" reset
    * @returns {boolean}
    */
-  get hasSoftReset() {
-    return (this.questions.length > 0 && this.resetConfig.questionsType === 'soft') || (this.presentationComponents.length > 0 && this.resetConfig.presentationComponentsType === 'soft');
+  get isSoftReset() {
```

Comment:
What about where `this.presentationComponents.length === 0` or ¬this.questions.length === 0`?

```suggestion
    const hasNoQuestions = (this.questions.length === 0);
    const hasNoPresentationComponents = (this.presentationComponents.length === 0);
    const hasQuestionsSoftReset = (hasNoQuestions || this.resetConfig.questionsType === 'soft');
    const hasPresentationComponentsSoftReset = (hasNoPresentationComponents  || this.resetConfig.presentationComponentsType === 'soft');
    const hasSoftReset = (hasQuestionsSoftReset && hasPresentationComponentsSoftReset);
    return hasSoftReset;
```